### PR TITLE
feat: confirmation and warnings for convert behavior

### PR DESCRIPTION
**Pull request by mshanemc**
> uses the prerelease of https://github.com/forcedotcom/source-deploy-retrieve/pull/1392 `12.3.0-qa.1`.  Before merging this, go back to SDR main (it's not required for this change, just helpful for QA) and remove the `yarn resolutions`

### What does this PR do?
adds a warning to make sure people don't have uncommited work before doing project modifications
adds a warning if people don't have main/default folders telling them that we're going to use them anyway
"reset" the sfProject Singleton instance to avoid stale presets during the conversion 

QA notes:
yarn resolutions is there temporarily to force STL to also use that SDR prerel
you can use this command to go from "normal" to CL beta or CL beta2, or from one beta to the other.

### What issues does this PR fix or reference?
@W-16486531@

### CODE REVIEW

File: messages/convert.source-behavior.md [hunk 1]
```diff
@@ -51,15 +51,13 @@ Your project has a default org (target-org) that uses source tracking. This oper
 - Run this command again.
 - Create a new org ("sf org create scratch" or "sf org create sandbox") and deploy the modified source.
 
-# error.packageDirectoryNeedsMainDefault
+# mainDefaultConfirmation
 
-The package directory %s doesn't have a main/default structure.
-This command moves metadata into a main/default structure, but your package directories aren't ready for it.
+- This command will put components in a newly created `main/default` folder in each package directory. You might need to re-organize them into your preferred structure.
```
Comment (jshackell-sfdc) [inline messages/convert.source-behavior.md:56]:
```suggestion
- This command puts components in a newly created `main/default` folder in each package directory. You might need to re-organize them into your preferred structure.
```

File: messages/convert.source-behavior.md [hunk 2]
```diff
@@ -51,15 +51,13 @@ Your project has a default org (target-org) that uses source tracking. This oper
...
 
-# error.packageDirectoryNeedsMainDefault.actions
+# basicConfirmation
 
-- Update %s to have all its metadata inside a main/default directory structure.
-- Run the command again.
+- This command will make changes to your project. Be sure you've committed any source changes before continuing so you can easily revert if necessary.
```
Comment (jshackell-sfdc) [inline messages/convert.source-behavior.md:60]:
```suggestion
- This command makes changes to your project. Be sure you've committed any source changes before continuing so you can easily revert if necessary.
```

**Review by jshackell-sfdc — APPROVED** [review]
Teeny edits. Thx.

**Review by shetzel — APPROVED** [review]

**Comment by mshanemc** [comment]
QA notes:

`apex run test`
✅ no changes when not using --concise
✅  hides passing test detail when all tests pass
✅ shows only failing tests when one is failing

`apex get test`
✅ shows only failing tests when one is failing

**Comment by shetzel** [comment]
QA Notes: I see the warnings, with dry-run and without.  Looks great!

**Pull request merged**